lib/models/device.ts

```diff
@@ -298,6 +305,35 @@ const getDeviceModel = function (
 		});
 	};
 
+	const historyTimeRangeFilterWithGuard = (
+		fromDate?: Date,
+		toDate?: Date,
+	): PineFilter<DeviceHistory> => {
+		let fromDateFilter;
+		let toDateFilter;
+		if (fromDate != null) {
+			if (fromDate instanceof Date) {
+				fromDateFilter = { $ge: fromDate };
+			} else {
+				throw new errors.BalenaInvalidParameterError('fromDate', fromDate);
```

Comment:
NIT: This could be using an early return pattern like `if (!(fromDate instanceof Date)) { throw ... }`.

Reply:
Yeah, will follow the fail early pattern